src/Instrumentation/ReactHTTP/src/ReactHttpInstrumentation.php

```diff
@@ -0,0 +1,161 @@
+            schemaUrl: 'https://opentelemetry.io/schemas/1.32.0',
+        );
+
+        /** @psalm-suppress UnusedFunctionCall */
+        hook(
+            Transaction::class,
+            'send',
+            pre: static function (Transaction $transaction, array $params, string $class, string $function, ?string $filename, ?int $lineno) use ($instrumentation): ?array {
+                $request = $params[0];
+
+                $propagator = Globals::propagator();
+                $parentContext = Context::getCurrent();
+
+                foreach ($propagator->fields() as $field) {
+                    $request = $request->withoutHeader($field);
+                }
+
+                /** @psalm-suppress ArgumentTypeCoercion */
+                $spanBuilder = $instrumentation
+                    ->tracer()
+                    // https://opentelemetry.io/docs/specs/semconv/http/http-spans/#http-client-span
+                    ->spanBuilder(sprintf('%s', $request->getMethod()))
+                    ->setParent($parentContext)
+                    ->setSpanKind(SpanKind::KIND_CLIENT)
+                    ->setAttribute(TraceAttributes::HTTP_REQUEST_METHOD, $request->getMethod())
+                    ->setAttribute(TraceAttributes::SERVER_ADDRESS, $request->getUri()->getHost())
+                    ->setAttribute(TraceAttributes::SERVER_PORT, $request->getUri()->getPort())
+                    ->setAttribute(TraceAttributes::URL_FULL, (string) $request->getUri())
+                    ->setAttribute(TraceAttributes::NETWORK_PROTOCOL_NAME, 'http')
+                    ->setAttribute(TraceAttributes::NETWORK_PROTOCOL_VERSION, $request->getProtocolVersion())
+                    ->setAttribute(TraceAttributes::HTTP_REQUEST_BODY_SIZE, $request->getHeaderLine('Content-Length'))
+                    ->setAttribute(TraceAttributes::URL_SCHEME, $request->getUri()->getScheme())
+                    ->setAttribute(TraceAttributes::USER_AGENT_ORIGINAL, $request->getHeaderLine('User-Agent'))
+                    // https://opentelemetry.io/docs/specs/semconv/code/
+                    ->setAttribute(TraceAttributes::CODE_FUNCTION_NAME, sprintf('%s::%s', $class, $function))
+                    ->setAttribute(TraceAttributes::CODE_FILE_PATH, $filename)
+                    ->setAttribute(TraceAttributes::CODE_LINE_NUMBER, $lineno);
+
+                /**
+                 * This cannot be tested with PHPUnit without modifying php.ini with:
+                 * otel.instrumentation.http.request_headers[]="Accept"
+                 */
+                // @codeCoverageIgnoreStart
+                foreach ((array) (get_cfg_var('otel.instrumentation.http.request_headers') ?: []) as $header) {
+                    if ($request->hasHeader($header)) {
+                        $spanBuilder->setAttribute(
+                            sprintf('%s.%s', TraceAttributes::HTTP_REQUEST_HEADER, strtolower($header)),
+                            $request->getHeaderLine($header)
```

Comment:
```suggestion
                            $request->getHeader($header)
```
Attribute has type `string[]`, same applies to response headers.

Reply:
Got it! Fixed in the next commit.